OpenRA.Mods.Common/Activities/Air/Land.cs

```diff
@@ -17,25 +19,40 @@ namespace OpenRA.Mods.Common.Activities
 {
 	public class Land : Activity
 	{
-		readonly Target target;
 		readonly Aircraft aircraft;
 		readonly WVec offset;
+		readonly int desiredFacing;
+		readonly CPos[] clearCells;
 
+		Target target;
+		WPos targetPosition;
+		WPos w1, w2, w3;
+		CPos landingCell;
 		bool landingInitiated;
-		bool soundPlayed;
+		bool finishedApproach;
 
-		public Land(Actor self, Target t, WVec offset)
+		public Land(Actor self, Target target = new Target(), int facing = -1, WVec offset = new WVec(), CPos[] clearCells = null)
 		{
-			target = t;
 			aircraft = self.Trait<Aircraft>();
+			this.target = target;
 			this.offset = offset;
+			this.clearCells = clearCells ?? new CPos[0];
+
+			// NOTE: desiredFacing = -1 means we should not prefer any particular facing and instead just
+			// use whatever facing gives us the most direct path to the landing site.
+			if (facing == -1 && aircraft.Info.TurnToLand)
```

Comment:
I had thought about it but decided it was better to have the `TurnToDock` case be handled by `ReturnToBase` instead and have it pass a facing to Land. This keeps the landing code more general and keeps everything that is specific to docking in `ReturnToBase.` I also think that for the future we should consider making the preferred dock facing (if any) a characteristic of the airfield/helipad instead of the aircraft. This would also imply unhardcoding the eastward landing direction for airfields.